Project structure

kappq opened the pull request:
I will keep this PR as a draft until further decisions on #45 will be made.

Closes #45 .

kappq committed:
Import from server instead of current package

kappq committed:
Separate Client and ConnectionManager in different files

LukeHankey commented on server/connection_manager.py:
```diff
@@ -0,0 +1,130 @@
+from server.events import EventResponse, ReplaceData
+
+
+@dataclass
+class RoomData:
+    """A dataclass for data about a specific room."""
+
+    owner_id: UUID
+    clients: set[Client]
+    code: str
+
+
+class ActiveRooms(TypedDict):
+    """A data structure for active rooms."""
+
+    name: str
+    data: RoomData
```
Can you replace all of this with:
```py
class RoomData(TypedDict):
    """A dataclass for data about a specific room."""

    owner_id: UUID
    clients: set[Client]
    code: str


ActiveRooms: TypeAlias = dict[str, RoomData]
```
This should fix the type issues. Remove the dataclass import and add `TypeAlias` to the typing import

kappq replied:
Fixed in 197e3aaf67a49d8ad35c73a61f679dbcf8807fc5.

kappq committed:
Change RoomData and ActiveRooms data structures

kappq committed:
Fix merge conflict

kappq committed:
Import from server instead of current package

LukeHankey commented on server/connection_manager.py:
```diff
@@ -0,0 +1,125 @@
+        """
+        self._rooms[room_code]["clients"].remove(client)
+
+        if len(self._rooms[room_code]["clients"]) == 0:
```
I think we can make this more pythonic: `if not len(...):`

kappq committed:
Use list as condition

This uses the list as a condition instead of checking if the length of the list is 0.
